packages/hdwallet-walletconnect-bridge/src/bridge.ts

```diff
@@ -0,0 +1,149 @@
+  }
+
+  async _onSessionRequest(error: Error | null, payload: WalletConnectSessionRequestPayload) {
+    this.log("Session Request", { error, payload });
+
+    const address = await this.wallet.ethGetAddress({ addressNList });
+    this.connector.approveSession({
+      chainId: payload.params[0].chainId ?? 4,
+      accounts: [address!],
+    });
+  }
+
+  async _onSessionUpdate(error: Error | null, payload: any) {
+    this.log("Session Update", { error, payload });
+  }
+
+  async _onConnect(error: Error | null, payload: any) {
+    this.log("Connect", { error, payload });
+  }
+
+  async _onDisonnect(error: Error | null, payload: any) {
+    this.log("Disonnect", { error, payload });
+  }
+
+  async _onCallRequest(error: Error | null, payload: WalletConnectCallRequest) {
+    this.log("Call Request", { error, payload });
+
+    let result: any;
+    switch (payload.method) {
+      // TODO: figure out what to do for "eth_sign" and "eth_signTypedData".
+      // MetaMaskHDWallet's ethSignMessage calls "personal_sign"
+      // and the other fns don't seem to be exposed on HDWallet
+      case "eth_sign": {
+        // result = await this.wallet.ethSignMessage({
+        //   addressNList,
+        //   message: payload.params[1],
+        // });
+        break;
+      }
+      case "eth_signTypedData": {
+        break;
+      }
+      case "personal_sign": {
+        const response = await this.wallet.ethSignMessage({
+          addressNList,
+          message: this.convertHexToUtf8IfPossible(payload.params[0]),
+        });
+        result = response?.signature;
+        break;
+      }
+      case "eth_sendTransaction": {
+        const tx = payload.params[0];
+        const response = await this.wallet.ethSendTx?.({
+          addressNList,
+          chainId: tx.chainId,
+          data: tx.data,
+          gasLimit: tx.gasLimit,
+          nonce: tx.nonce,
+          to: tx.to,
+          value: tx.value,
+        });
+        result = response?.hash;
+        break;
+      }
+      case "eth_signTransaction": {
+        const tx = payload.params[0];
+        const response = await this.wallet.ethSignTx({
+          addressNList,
+          chainId: tx.chainId,
+          data: tx.data,
+          gasLimit: tx.gas,
+          nonce: tx.nonce,
+          to: tx.to,
+          value: tx.value,
+        });
+        result = response?.serialized;
+      }
+    }
+
+    if (result) {
+      this.log("Approve Request", { payload, result });
+      this.connector.approveRequest({ id: payload.id, result });
+    } else {
+      const message = "JSON RPC method not supported";
+      this.log("Reject Request", { payload, message });
+      this.connector.rejectRequest({ id: payload.id, error: { message } });
+    }
+  }
+
+  private log(_eventName: string, _properties: object) {
+    if (process.env.NODE_ENV !== "test") {
+      // TODO: what are conventions around log statements (which can be very helpful in dev mode)
```

Comment:
Here, it's probably a good idea to use the [@shapeshiftoss/logger](https://www.npmjs.com/package/@shapeshiftoss/logger) package. This is used in the other ShapeShift repositories but hasn't been brought into hdwallet yet.